module-apps/apps-common/ApplicationCommon.cpp

```diff
@@ -914,35 +920,46 @@ namespace app
     void ApplicationCommon::abortPopup(gui::popup::ID id)
     {
         const auto popupName = gui::popup::resolveWindowName(id);
-        LOG_INFO("abort popup: %s from window %s", popupName.c_str(), getCurrentWindow()->getName().c_str());
-        if (not windowsStack().pop(popupName)) {
+
+        /* Hack for dropping tethering confirmation window on locked phone to prevent removing
+         * phone lock popup, thus unlocking the phone without providing passcode. */
+        if ((id == gui::popup::ID::Tethering) &&
+            windowsStack().isWindowOnStack(gui::popup::window::phone_lock_window)) {
+            LOG_INFO("Aborting tethering confirmation window on locked phone");
+            windowsStack().drop(popupName);
+            return;
+        }
+
+        /* Regular popup aborting handler */
+        LOG_INFO("Aborting popup %s from window %s", popupName.c_str(), getCurrentWindow()->getName().c_str());
+        if (!windowsStack().pop(popupName)) {
             return;
         }
         if (popupName == getCurrentWindow()->getName()) {
             returnToPreviousWindow();
         }
     }
 
-    void ApplicationCommon::requestShutdownWindow(std::string windowName)
+    void ApplicationCommon::requestShutdownWindow(const std::string &windowName)
     {
 #if DEBUG_APPLICATION_MANAGEMENT == 1
         LOG_INFO("switching [%s] to shutdown window: %s",
                  GetName().c_str(),
                  windowName.length() ? windowName.c_str() : default_window.c_str());
 #endif
-        if (not windowsFactory.isRegistered(windowName)) {
+        if (!windowsFactory.isRegistered(windowName)) {
             LOG_ERROR("Cannot find window %s windowsFactory in application: %s", windowName.c_str(), GetName().c_str());
             return;
         }
 
         pushWindow(windowName);
         const auto window = getWindow(windowName);
-        if (not window) {
+        if (window == nullptr) {
```

Comment:
Usually, you use `!` so maybe also here ;)

Reply:
I try to explicitly compare pointers against `nullptr` value, so that it is easily seen that we're dealing with pointer. Using `!` creates an ambiguity - it can be used to check either null pointer, empty `std::optional` or `bool` with `false` value. But that's just my personal preference, so if you think `!` fits better here I'll change it ;)